cli/util/uploader.go

```diff
@@ -0,0 +1,352 @@
+	if err != nil {
+		return fmt.Errorf("failed to update index files after upload: %w", err)
+	}
+	ctx.App.Writer.Write([]byte("Upload completed successfully.\n"))
+	return nil
+}
+
+// fetchMaxBlockIndex searches for the maximum block index in the container.
+func fetchMaxBlockIndex(ctx context.Context, clientSDK *client.Client, containerID cid.ID, priv *keys.PrivateKey, currentHeight uint, attributeKey string) (int, error) {
+	height := int(currentHeight)
+	var (
+		finalResult int
+		finalErr    error
+	)
+
+	searchBatch := func(start, end int) (int, bool, error) {
+		prm := client.PrmObjectSearch{}
+		filters := object.NewSearchFilters()
+		filters.AddFilter(attributeKey, fmt.Sprintf("%d", start), object.MatchNumGE)
+		filters.AddFilter(attributeKey, fmt.Sprintf("%d", end), object.MatchNumLE)
+		prm.SetFilters(filters)
+
+		objectIDs, err := neofs.ObjectSearch(ctx, clientSDK, priv, containerID.String(), prm)
+		if err != nil {
+			return 0, false, fmt.Errorf("failed to search objects from %d to %d: %w", start, end, err)
+		}
+
+		numOIDs := len(objectIDs)
+		fmt.Printf("Found %d blocks between %d and %d", numOIDs, start, end)
+
+		if numOIDs == 0 {
+			return 0, false, nil // Keep searching.
+		}
+
+		// Return the max block index found in this batch
+		maxInBatch := start + numOIDs - 1
+
+		// Stop immediately after finding the first non-empty batch
+		return maxInBatch, true, nil
+	}
+
+	for ; height > 0; height -= searchBatchSize {
+		startIndex := height - searchBatchSize + 1
+		if startIndex < 0 {
+			startIndex = 0
+		}
+
+		maxInBatch, foundNonEmptyBatch, err := searchBatch(startIndex, height)
+		if err != nil {
+			finalErr = err
+			break
+		}
+
+		if maxInBatch > finalResult {
+			finalResult = maxInBatch
+		}
+
+		// Stop the search as soon as we find any valid blocks
+		if foundNonEmptyBatch {
+			break
+		}
+	}
+
+	if finalErr != nil {
+		return 0, finalErr
+	}
+
+	return finalResult, nil
+}
+
+func fetchBlockData(rpcClient *rpcclient.Client, index uint) ([]byte, error) {
+	if index%1000 == 0 {
+		fmt.Println("Fetching block", index)
+	}
+	block, err := rpcClient.GetBlockByIndex(uint32(index))
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch block %d: %w", index, err)
+	}
+
+	bw := io.NewBufBinWriter()
+	block.EncodeBinary(bw.BinWriter)
+
+	if bw.Err != nil {
+		return nil, fmt.Errorf("failed to encode block %d: %w", index, bw.Err)
+	}
+
+	return bw.Bytes(), nil
+}
+
+func uploadBObjWithSlicer(ctx context.Context, clientSDK *client.Client, account wallet.Account, containerID cid.ID, objData []byte, attributeKey, attributeValue string, sessionToken *session.Object) error {
+	signer := user.NewAutoIDSignerRFC6979(account.PrivateKey().PrivateKey)
+	var ownerID user.ID
+	ownerID.SetScriptHash(account.PrivateKey().GetScriptHash())
+
+	slc, err := slicer.New(ctx, clientSDK, signer, containerID, ownerID, sessionToken)
+	if err != nil {
+		return fmt.Errorf("failed to create slicer: %w", err)
+	}
```

Comment:
Create signer and ownerID once reuse it afterwards. Don't use slicer, upload directly without it, our objects are small.